### ISSUE TITLE
Eurotronic SPZB0001 eurotronic_error_status: 1, softwareBuildID: 0000000

### ISSUE THREAD

**Issue by prankousky** [Koenkk/zigbee-herdsman-converters#988]
Hi everybody,

I have one particular `SPZB0001` thermostat (off of many that I use) that never works as expected. I just tried debugging this a bit via `mqtt`, and realized: this device does not have a `softwareBuildID`, and it has `eurotronic_error_status: 1` set.

It *seems* like I can turn the device completely off or to on/automatic, but when I change the temperature, nothing will happen. I compared it to another device that I know works fine, see table below. The order is different, too. Each key on the very left is the order of `Jonna/Heizung`; `Schlafzimmer/Heizung`, however, has a completely different order. I just realized that after starting the table, so I adapted the values accordingly, thought I should still mention on.

| | Jonna (not working)           | Schlafzimmer (working)    |                           |   |   
|-------------------------------|---------------------------|---------------------------|---|---|
| pi_heating_demand             | 0                         | 35                        |   |   |
| linkquality                   | 72                        | 33                        |   |   |
| last_seen                     | 2020-02-04T12:32:11+01:00 | 2020-02-04T12:21:43+01:00 |   |   |
| current_heating_setpoint      | 17.5                      | 16.5                      |   |   |
| local_temperature             | 20                        | 14.44                     |   |   |
| **eurotronic_error_status**       |**1**                         | **0**                         |   |   |
| occupied_heating_setpoint     | 17                        | 21                        |   |   |
| unoccupied_heating_setpoint   | 16                        | 16                        |   |   |
| eurotronic_host_flags (below) |                           |                           |   |   |
| mirror_display                | false                     | false                     |   |   |
| boost                         | false                     | false                     |   |   |
| window_open                   | false                     | false                     |   |   |
| child_protection              | false                     | false                     |   |   |
| system_mode                   | auto                      | auto                      |   |   |
| **eurotronic_system_mode**        | **1**                         | **0**                         |   |   |
| battery                       | 10                        | 80                        |   |   |
| device (below)                |                           |                           |   |   |
| friendly_name                 | Jonna/Heizung             | Schlafzimmer/Heizung      |   |   |
| model                         | SPZB0001                  | SPZB0001                  |   |   |
| ieeeAddr                      | &lt;ieee&gt;              | &lt;ieee&gt;              |   |   |
| networkAddress                | &lt;net&gt;               | &lt;net&gt;               |   |   |
| type                          | EndDevice                 | EndDevice                 |   |   |
| manufacturerID                | 4151                      | 4151                      |   |   |
| manufacturerName              | Eurotronic                | Eurotronic                |   |   |
| powerSource                   | Battery                   | Battery                   |   |   |
| **applicationVersion**            | **22**                        | **15**                        |   |   |
| stackVersion            | 5                             | 5                         |                           |   |   |
| zclVersion                    | 2                         | 2                         |   |   |
| **hardwareVersion**               | **35**                        | **34**                        |   |   |
| **dateCode**                      | **20191014**                  | **20181205**                  |   |   |
| **softwareBuildID**               | **00000000**                  | **15181120**                  |   |   |

So there are quite some differences, which I marked in bold text. I do not have older output saved, but I know for sure that this device **used to** work just fine. I was able to control temperature remotely and get the desired temperature update on the thermostat as well as change temperature on the thermostat and get a quick status update in my UI (using Home Assistant).

As you can see here, when changing temperature, nothing will happen

```
zigbee2mqtt/Jonna/Heizung/set/current_heating_setpoint 17.0
zigbee2mqtt/Jonna/Heizung/set/system_mode off
zigbee2mqtt/Jonna/Heizung {"pi_heating_demand":0,"linkquality":69,"last_seen":"2020-02-04T12:32:11+01:00","current_heating_setpoint":17.5,"local_temperature":20,"eurotronic_error_status":1,"occupied_heating_setpoint":17,"unoccupied_heating_setpoint":16,"eurotronic_host_flags":{"mirror_display":false,"boost":false,"window_open":false,"child_protection":false},"system_mode":"auto","eurotronic_system_mode":1,"battery":10,"device":{"friendlyName":"Jonna/Heizung","model":"SPZB0001","ieeeAddr":"0x00158d0001ffc6d1","networkAddress":36230,"type":"EndDevice","manufacturerID":4151,"manufacturerName":"Eurotronic","powerSource":"Battery","applicationVersion":22,"stackVersion":5,"zclVersion":2,"hardwareVersion":35,"dateCode":"20191014","softwareBuildID":"00000000"}}
```

What can I do to fix this? I have "solved" this problem before by buying a different SPZB0001 thermostat and exchanging it with this one. It worked fine for quite some time. Suddenly it stopped working, so I assume it can be fixed to work fine again. I just don't know how.

Thank you for your support :)

**Comment by sjorge** [comment]
Odd, bit 0,1 and 2 are reserve in the bitmap for the error so not sure what error 1 is suppose to be.
Have you tried removing the device form the valve and reinstalling it (including joining it?)

**Comment by prankousky** [comment]
So far, I only factory reset the device by removing the batteries, inserting them, pressing all the buttons, and initiating the pairing process - but without removing it from the valve. I will try that now and  report back, if there is any change.

**Comment by allofmex** [comment]
> "softwareBuildID":"00000000"

I see this frequently to on some of mine. It's usually enough just to replug batteries. No factory reset needed.

Guess it's a firmware issue in spz0001

**Comment by prankousky** [comment]
Removing from the valve did nothing. I removed it, reset it, re-paired it. Thermostat is set to 19°C. Home Assistant displays it as 15.5°C (has been on that value for days). When I change the thermostat itself, Home Assistant will stay at 15.5°C. When I change the value in Home Assistant, it **will** change the thermostat, but not confirm the value (as soon as I reload the page, it will display 15.5°C again, even though the change made by Home Assistant had successfully been transmitted to the thermostat, as it does change to the newly set value).

@allofmex I usually just replug the batteries as well when there are minor issues. But not reporting the current temperature back made me thing I had to factory reset and re-pair the device to fix it. I deleted it from my config as well. But so far, nothing worked. 

Last time I had this issue, I ordered a new thermostat and returned the old one - BUT that new thermostat (that used to work) produces the very same issue again now, which makes me doubt that this is related to the individual thermostat and must be either caused by communication between zigbee2mqtt and the device, or the device itself (firmware issue, for example, as you said).

Either way, this is very, very frustrating.

**Comment by sjorge** [comment]
> Either way, this is very, very frustrating.

This is my geenraly feeling when dealing with my SPZB0001, then again I have just as much issues with my Schneider Electric one. 🤷‍♂

**Comment by prankousky** [comment]
Do you know of any thermostats available in Germany (or at least within Europe) that **do** work reliably? I used to have the eq3 Maxcube and those thermostats, which you could only control so many times per hour, then (because of some rf regulations??) it would not transmit for a certain amount of time.

Switching to SPZB0001 was such a relief, but now it is causing problems over and over again. I don't care if it is zigbee, ideally it'd just be an open source wifi thermostat, but I doubt that this exists. I have one DECT thermostat that works with the `fritzbox`, which only updates every 10 minutes. It does work fine, but I cannot wait 9:59 minutes in the worst case scenario for my thermostats to update each time I change a value o.O

**Comment by allofmex** [comment]
> But not reporting the current temperature back made me thing I had to factory reset and re-pair

Yes, that's not related to the sw 000000 issue.

Sounds again like `reporting` does not get successful configured. What do you logs say about `device configured`?

**Comment by sjorge** [comment]
@prankousky not really :(

I have these:
- https://www.zigbee2mqtt.io/devices/SPZB0001.html
- https://www.zigbee2mqtt.io/devices/WV704R0A0902.html

But overal, I find the SPZB0001 best of the two, as on the other one you do not get manual control and there are still some integration issues. But the hardware does feel better.

SPZB0001: flaky firmware(-), works well with zigbee2mqtt(+), manual control(+)
WV704R0A0902: no manual control(-), still some issues with zigbee2mqtt(-), unsure on firmware... some odd issues with it resetting setpoint after 1h and such :/

**Comment by allofmex** [comment]
Error 1 is `Valve adaption failed`
https://www.zigbee2mqtt.io/devices/SPZB0001.html
Do you still have this error? In that case this should be fixed first. Else the thermostats cannot work properly. Please check your plastic valve adapter.

(I have 6 Eurotronic, and for me they work more or less reliable. Some issues remain, but it's useable. I don't have alternative device suggest)

**Comment by prankousky** [comment]
These two images are of the same thermostat. It will always *report* `15.5` as temperature. When I change the value, it will adapt, but not report. This one does not have the `Valve adaption failed` error, but I will check on that one now as well.

![grafik](https://user-images.githubusercontent.com/11297070/74828367-47a79000-530f-11ea-8fee-cce743f02c18.png)
![grafik](https://user-images.githubusercontent.com/11297070/74828441-686fe580-530f-11ea-830b-d418bc45b01b.png)

The other thermostat thermostat with the valve adaption error has not yet reported its status. I have replaced the battery, which was at 10%. While it seems to have paired without a problem, I am not receiving updates on its status yet. Will report back once those come in.

I added these lines to my `configuration.yaml` and restarted, but no difference can be noticed at the moment.

```
mqtt:
  include_device_information: true
advanced:
  log_level: debug
  cache_state: true
  report: true
  rtscts: true # no idea what this even is, and it is true by default, but I included it anyway as I am running out of ideas 
  last_seen: 'ISO_8601_local'
  elapsed: true
  timestamp_format: 'YYYY-MM-DD HH:mm:ss'
```

**Comment by prankousky** [comment]
I changed my `channel`, `network_key`, and `pan_id` and re-paired every single device. As far as I can tell at  the moment, everything works fine. At least everything I tested worked, including those thermostats that did not previously do what they were supposed to. 

So while I'd say this is good, I still don't know a decent solution. One cannot always change their entire network when some device fails... what else could be done, though?
